### DISCUSSION TITLE
ios App freeze about 5-10 Seconds after start

### DISCUSSION THREAD

**Discussion by Solear** [immich-app/immich#14770] [Q&A]
Hello! I am a passionate Immich user and supporter.
Immich runs via Docker, there are about 240 GB of photos stored. 
The iOS app also works great. Except: every time I open the app, it freezes for 7-8 seconds after approx. 2 seconds. After that, everything works normally. Regardless of whether I am in the WLAN or via Tailscale, whether I have stored the IP or the domain for the Immich server in the app.

Can you please point to the right direction?

![5985645915140113035](https://github.com/user-attachments/assets/586beb91-ab9b-4797-b4d2-49d39c2216ac)

**Comment by alextran1502** [comment]
Hello, do you mind let me know what is your iPhone model?

**Reply by Solear** [reply]
Hello, thank you for your respone,  its an iphone 15 Pro 256GB, iOS 18.2 (beta). But its the same freeze with the non-beta.

**Comment by sbogomolov** [comment]
I have a similar issue. There are two users, first one has ~11k assets and things work fine. The second user has ~100k assets (not backed up yet). An app on iPhone freezes a few seconds after startup and never unfreezes.

**Reply by alextran1502** [reply]
Are both users experiencing the same symptom? Can you share the mobile app log? How long have both users used the app?

**Reply by sbogomolov** [reply]
Only user with ~100k assets experiences the issue. For use with ~10k assets things work well. All assets were backed up and there are no freezes.

The user with ~100k assets experiences these freezes from the beginning. When we started the app for the first time we were able to add Recents (containing all assets) to the list of albums to backup, and then app froze. It freezes every time a few seconds after the app is started since then. I'm not sure how to get logs from the app since it does not unfreeze.

**Reply by alextran1502** [reply]
Thank you, are the 100000 assets all from iCloud

**Reply by sbogomolov** [reply]
All of the assets are backed up to iCloud. I assume that some of them might be automatically offloaded there to conserve space (there is this option to do this transparently).

**Reply by VincenzoKoestler** [reply]
@sbogomolov I am just experiencing the same but realized after restarting the app the library contains more pictures. So I think the freeze is just because it’s overwhelmed with processing that many pictures. I’ll just keep the app open and see what happens. Where you able to figure out anything different?

**Reply by sbogomolov** [reply]
@VincenzoKoestler no, I have not found a better solution. It would sometimes unfreeze and upload a few photos. That's all I ever got :)

**Comment by bradley-varol** [comment]
I also have this issue on my 13 Pro Max. It always freezes on first open and also always when switching to the app - both after 2-3s. It happens even when in the app settings. Switching apps while it’s frozen and then back again a few minutes later, it is still frozen for 1-2s, then it unfreezes for 2-3s and freezes again as if I had just opened the app. Something is blocking the rendering thread I would assume 😅

My assets are currently backed up to iCloud (while I transition to Immich), and most are available on the device (ie not offloaded).

Let me know if you need any more information or if a screen recording would be useful!

**Reply by alextran1502** [reply]
How many assets do you have on iCloud?

**Reply by bradley-varol** [reply]
> How many assets do you have on iCloud?

290GB - 25,000 photos, 4,000 videos

**Comment by Solear** [comment]
I think could solve this freeze Problem temporary: immich App, settings, delete duplicates (it was over 40k, my Library). 
After that there was no freezing (Checkmark in the cloud symbol)

…but now its ersynching again and its freezing again. I think its always freezing if there is a synching. But it never end with syncing because some errors (see picture in No1)

**Comment by Solear** [comment]
I figured out another problem (possibly related to the freeze): the battery drain. On my iPhone 15 Pro immich is in the top 3 list for battery drain (background sync). more than an open insta-app for an hour. 
In the past there was no battery drainage problem with immich. 

Its look like the freezing sync drains some battery in the background - repeadly. 
(Immich via webbrowser is seamles fast, its only the ios app).

Let me know if you need some special logs. But today I'm heading to vacation for one week - please understand my delayed responses.

**Comment by bradley-varol** [comment]
I also find my iPhone is not automatically backing up my photos in the
background. If I don't open the Immich app on my phone for a few days and
open Immich in a browser, i'll see that the last photos are from a few days
ago.

I do have "Background App Refresh" and "Foreground App Refresh"(?) enabled.

**Comment by Solear** [comment]
Any news on this? I cant proper use immich on ios because of this. I already set up immich from scratch new and synch over ios.

**Reply by bradley-varol** [reply]
Same here. I absolutely love Immich but it's not reliable on my iPhone. I don't want to have to check every few days if it's backed up or not, and the freezing is annoying. I've moved back to iCloud for the time being and hope to return to Immich soon!

**Comment by H4usi** [comment]
I have the same problem. 

iPhone 14 Pro Max 512GB (iOS 18.4)
Images: 35.198
Duplicates: 461
Albums: 61

I have tried to show the problem via the log, unfortunately even with activated troubleshooting and the log level set to Finest it is not possible to see why this is happening. The log also just stops for a few seconds. Also always at different positions while it synchronises the albums (they are never the same albums) 

Is there anything that could be done to track down the problem?

**Reply by alextran1502** [reply]
@H4usi how many photos/videos do you have on iCloud?

**Reply by H4usi** [reply]
Photos: 29.139
Videos: 949

**Comment by Atlas553** [comment]
Same here with a big iCloud library. 

![image](https://github.com/user-attachments/assets/bffbd2a9-1fd6-4808-97e3-941acd592bda)

**Reply by PhilipSchaefer86** [reply]
Same here. iPhone 15pro (iOS 18.4.1), with an iCloud library of 53k photos and 1300 videos.

**Reply by felixwaage** [reply]
I got the same issue with iPhone 12 Pro and 30k items in my iCloud library.

**Comment by RomainBrgnz** [comment]
Hello, I'm experiencing the same problem. I have two accounts: the first one works fine (17k assets), but the second one isn't working (33k assets) the app freezes about 5 seconds after launching. I'm on the last release.

**Comment by alextran1502** [comment]
Please see #19370 for more information
